beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -44,32 +44,39 @@ func (r *Service) validateAggregateAndProof(ctx context.Context, pid peer.ID, ms
 		traceutil.AnnotateError(span, err)
 		return false
 	}
-	m, ok := raw.(*ethpb.AggregateAttestationAndProof)
+	m, ok := raw.(*ethpb.SignedAggregateAttestationAndProof)
 	if !ok {
 		return false
 	}
 
+	// Verify this is the first aggregate received from the aggregator with index and slot.
+	if r.hasSeenAggregatorIndexSlot(m.Message.Aggregate.Data.Slot, m.Message.AggregatorIndex) {
+		return false
+	}
+
 	// Verify aggregate attestation has not already been seen via aggregate gossip, within a block, or through the creation locally.
-	seen, err := r.attPool.HasAggregatedAttestation(m.Aggregate)
+	seen, err := r.attPool.HasAggregatedAttestation(m.Message.Aggregate)
 	if err != nil {
 		traceutil.AnnotateError(span, err)
 		return false
 	}
 	if seen {
 		return false
 	}
-	if !r.validateBlockInAttestation(ctx, m) {
+	if !r.validateBlockInAttestation(ctx, m.Message) {
 		return false
 	}
 
-	if !r.validateAggregatedAtt(ctx, m) {
+	if !r.validateAggregatedAtt(ctx, m.Message) {
 		return false
 	}
 
-	if !featureconfig.Get().DisableStrictAttestationPubsubVerification && !r.chain.IsValidAttestation(ctx, m.Aggregate) {
+	if !featureconfig.Get().DisableStrictAttestationPubsubVerification && !r.chain.IsValidAttestation(ctx, m.Message.Aggregate) {
 		return false
 	}
 
+	r.setAggregatorIndexSlotSeen(m.Message.Aggregate.Data.Slot, m.Message.AggregatorIndex)
```

Comment:
Shouldn't this be done in the subscriber?

Reply:
agree